sygnal/webpushpushkin.py

```diff
@@ -104,26 +113,42 @@ def __init__(self, name, sygnal, config):
             self.vapid_private_key = Vapid.from_file(private_key_file=privkey_filename)
         except VapidException as e:
             raise PushkinSetupException("invalid 'vapid_private_key' file") from e
-        vapid_contact_email = self.get_config("vapid_contact_email")
-        if not vapid_contact_email:
+        self.vapid_contact_email = self.get_config("vapid_contact_email")
+        if not self.vapid_contact_email:
             raise PushkinSetupException("'vapid_contact_email' not set in config")
-        self.vapid_claims = {"sub": "mailto:{}".format(vapid_contact_email)}
 
     async def _dispatch_notification_unlimited(self, n, device, context):
         p256dh = device.pushkey
         if not isinstance(device.data, dict):
             logger.warn(
-                "device.data is not a dict for pushkey %s, rejecting pushkey", p256dh
+                "Rejecting pushkey %s; device.data is not a dict", device.pushkey
             )
             return [device.pushkey]
 
         endpoint = device.data.get("endpoint")
         auth = device.data.get("auth")
+        endpoint_domain = urlparse(endpoint).netloc
+        if self.allowed_endpoints:
+            match = next(
+                (
+                    regex
+                    for regex in self.allowed_endpoints
+                    if regex.fullmatch(endpoint_domain)
+                ),
+                None,
+            )
```

Comment:
This is pretty difficult to read, I think you want something more like:

```python
allowed = any(regex.fullmatch(endpoint_domain) for regex in self.allowed_endpoints)
```

Reply:
that is way better indeed, only came up with that monstrosity after looking online how to to do a `find`